src/jvmTest/java/it/krzeminski/snakeyaml/engine/kmp/comments/EmitterWithCommentEnabledTest.kt

```diff
@@ -0,0 +1,639 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package it.krzeminski.snakeyaml.engine.kmp.comments
+
+import it.krzeminski.snakeyaml.engine.kmp.api.DumpSettings
+import it.krzeminski.snakeyaml.engine.kmp.api.Load
+import it.krzeminski.snakeyaml.engine.kmp.api.LoadSettings
+import it.krzeminski.snakeyaml.engine.kmp.api.StreamDataWriter
+import it.krzeminski.snakeyaml.engine.kmp.common.FlowStyle
+import it.krzeminski.snakeyaml.engine.kmp.common.ScalarStyle
+import it.krzeminski.snakeyaml.engine.kmp.composer.Composer
+import it.krzeminski.snakeyaml.engine.kmp.emitter.Emitter
+import it.krzeminski.snakeyaml.engine.kmp.events.*
+import it.krzeminski.snakeyaml.engine.kmp.parser.ParserImpl
+import it.krzeminski.snakeyaml.engine.kmp.scanner.StreamReader
+import it.krzeminski.snakeyaml.engine.kmp.serializer.Serializer
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+import java.io.StringWriter
+
+class EmitterWithCommentEnabledTest {
+    private fun runEmitterWithCommentsEnabled(data: String): String {
+        val output: StreamDataWriter = MyWriter()
+
+        val dumpSettings = DumpSettings.builder()
+            .setDefaultScalarStyle(ScalarStyle.PLAIN)
+            .setDefaultFlowStyle(FlowStyle.BLOCK)
+            .setDumpComments(true)
+            .build()
+        val serializer = Serializer(dumpSettings, Emitter(dumpSettings, output))
+
+        serializer.emitStreamStart()
+        val loadSettings = LoadSettings.builder()
+            .setParseComments(true)
+            .build()
+        val composer = Composer(
+            loadSettings,
+            ParserImpl(loadSettings, StreamReader(loadSettings, data))
+        )
+        for (node in composer) {
+            serializer.serializeDocument(node)
+        }
+        serializer.emitStreamEnd()
+
+        return output.toString()
+    }
+
+    private fun producePrettyFlowEmitter(output: StreamDataWriter): Emitter {
+        val dumpSettings = DumpSettings.builder()
+            .setDefaultScalarStyle(ScalarStyle.PLAIN)
+            .setDefaultFlowStyle(FlowStyle.FLOW)
+            .setDumpComments(true)
+            .setMultiLineFlow(true)
+            .build()
+        return Emitter(dumpSettings, output)
+    }
+
+    @Test
+    fun testEmpty() {
+        val data = ""
+
+        val result = runEmitterWithCommentsEnabled(data)
+        assertEquals(data, result)
+    }
+
+    @Test
+    fun testWithOnlyComment() {
+        val data = "# Comment\n\n"
+
+        val result = runEmitterWithCommentsEnabled(data)
+        assertEquals(data, result)
+    }
+
+    @Test
+    fun testCommentEndingALine() {
+        val data = """key: # Comment
+  value
+"""
```

Comment:
suggestion: we could use `trimIndent()` function to make the input data more readable

Reply:
Ah right, in these cases it can be used as well! Let me do it.

Reply:
From my quick attempt, it was challenging to make it right when it comes to e.g. trailing new lines. Also if we want to preserve exactly these strings, it's challenging because there's no new line at the beginning of some of them, and using the `.trimIndent()` approach is impractical. Let's merge it as is, we can always revisit especially when converting other tests.